### ISSUE TITLE
Actor.CreateFrom - API is unreliable when called from onStarted() handler

### ISSUE THREAD

**Issue by willneedit** [Microsoft/mixed-reality-extension-sdk#87]
An initial rotation as passed in the transform in Actor.CreateFromGLTF or Actor.CreateFromLibrary is ignored. Scaling and positioning works fine, but rotation has no effect.

UPDATE: Rotation gets ignored if one creates a bunch of similar objects in one function call (like a circle of candles or standing stones) and doesn't wait for the ForwardPromise<Actor> getting fully resolved.

WORKAROUND: Use 

> actorVar = await Actor.CreateFromLibrary(...)

 rather than 

> actorVar = Actor.CreateFromLibrary(...).value

.

Related comment: https://altspacevrsdk.slack.com/archives/C0M6J4YBG/p1545414641022300

**Comment by stevenvergenz** [comment]
If I'm reading our loader code right, we generate a unique ID node-side for each call, but the actual GameObject unity-side never gets associated with it, it gets generated a second ID after it's loaded. So updates to the original ID don't affect the Unity scene.

**Comment by eanders-MS** [comment]
@willneedit Can you provide sample code for repro?

**Comment by willneedit** [comment]
```
      	let i = 0;
	for (i = 0; i < 360; i = i + 40) {
	    const triggerPromise = Actor.CreatePrimitive(this.context,
            {
                definition: { shape: PrimitiveShape.Box, dimensions: new Vector3(0.5, 0.5, 0.5)},
                actor: {
                    name: 'test ' + i,
		    transform: { rotation: Quaternion.RotationAxis(Vector3.Right(), i * DegreesToRadians) }
                }
            });
	    const trigger = triggerPromise.value; // or await triggerPromise
	}
```

Expected: A star-shaped object consisting of nine boxes, overlaying each other and evenly rotated
Got: What looks like two boxes, one at a fourty degrees angle rotated in relation to the other one.

**Comment by eanders-MS** [comment]
@willneedit I copied your code verbatim and it appears to have worked correctly. Here's what I see in Unity:
![image](https://user-images.githubusercontent.com/12176807/50377385-44acf480-05d1-11e9-9998-59414096cdca.png)
Here it is in Altspace:
![image](https://user-images.githubusercontent.com/12176807/50377423-ff3cf700-05d1-11e9-9995-36f5699a6ddd.png)

Does that look right to you?

**Comment by willneedit** [comment]
Looks pretty much correct to me. I'll have to investigate further...

**Comment by eanders-MS** [comment]
Ok. Let us know what you find. If you'd like a second pair of eyes on your project I'd be happy to take a gander.

**Comment by willneedit** [comment]
For another object, I try to create a keyboard using the following code snippet:

```
        for (i = 0; i < 39; i++) {
            const xpos = (i % 8) * 0.10 + -0.35;
            const ypos = Math.trunc(i / 8) * -0.10 + 0.20;

            const key = Actor.CreatePrimitive(this.context,
                {
                    definition: { shape: PrimitiveShape.Box, dimensions: new Vector3(0.09, 0.09, 0.01)},
                    addCollider: true,
                    actor: {
                        // parentId: rootNode.id,
                        transform: { position: { x: xpos, y: ypos, z: 0 }},
                        name: 'button ' + i
                    }
                }).value;
            key.setBehavior(ButtonBehavior).onClick('pressed', (userId: string) => this.keypressed(i));
        }

```

Basically, place an eight by five panel of buttons, with the one on the lower right being left out. When I trigger the function from onUserJoined(), everything pans out okay. But when used from onStarted(), this function (seemingly) creates nothing.

Replacing **const key** with **const keyPromise** and then awaiting on the promise draws all the keys but the first when used in onStarted().

In addition I had used a root transformation (see parentId property), but when I instantiated an empty actor and awaited on this, the root transformation was missing and the first key was drawn.

So, in essence, even when trying to sync things, the very first instantiated actor falls under the table, when used from onStarted() rather than onUserJoined().

Not sure if it rates as a bug but rather a caveat, to think twice about building up any geometry in onStarted() but wait for it until the first user announced itself in onUserJoined(). Worse if it's a timing issue, and that may account for the spurious problems with the HelloWorld example.
